account-kit/smart-contracts/src/ma-v2/account/nativeSMASigner.ts

```diff
@@ -47,7 +51,55 @@ export const nativeSMASigner = (
   accountAddress: Address,
   deferredActionData?: Hex
 ) => {
+  const signingMethods: SigningMethods = {
+    prepareSign: async (
+      request: SignatureRequest
+    ): Promise<SignatureRequest> => {
+      let ret: SignatureRequest = {
+        type: "eth_signTypedData_v4",
+        data: {
+          domain: {
+            chainId: Number(chain.id),
+            verifyingContract: accountAddress,
+          },
+          types: {
+            ReplaySafeHash: [{ name: "hash", type: "bytes32" }],
+          },
+          message: {
+            hash: "",
+          },
+          primaryType: "ReplaySafeHash",
+        },
+      };
+
+      if (request.type === "personal_sign") {
+        ret.data.message.hash = hashMessage(request.data);
+      } else if (request.type === "eth_signTypedData_v4") {
+        const isDeferredAction =
+          request.data?.primaryType === "DeferredAction" &&
+          request.data?.domain?.verifyingContract === accountAddress;
+
+        if (isDeferredAction) {
+          ret.data = request.data;
+        } else {
+          ret.data.message.hash = hashTypedData(request.data);
+        }
+      } else {
+        throw new Error(`Unsupported signature request type`);
```

Comment:
nit: could use `assertNever` here to guard against us ever adding more types & forgetting to cover them here.